bigbluebutton-html5/imports/ui/components/plugins-engine/data-creation/mutation-trigger-handler/hooks.ts

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { MutationResult } from '@apollo/client';
+import { useState, useEffect } from 'react';
+import { isEqual } from 'radash';
+import usePreviousValue from '/imports/ui/hooks/usePreviousValue';
+import { DataCreationHookEnums } from 'bigbluebutton-html-plugin-sdk/dist/cjs/data-creation/enums';
+import { UpdatedEventDetails } from 'bigbluebutton-html-plugin-sdk/dist/cjs/core/types';
+import { MutationResultObject } from 'bigbluebutton-html-plugin-sdk/dist/cjs/data-creation/types';
+import { HookEvents } from 'bigbluebutton-html-plugin-sdk/dist/cjs/core/enum';
+import projectMutationResult from './utils';
+
+/**
+ * This hook sends the projected mutation result back to the plugin.
+ * It observes only the relevant part of the GraphQL result and forwards
+ * it when changes occur, avoiding unnecessary data
+ *
+ * @param resultFromGraphql result of the useMutation from graphql with lots of information
+ * @param mutation mutation used to create the trigger function
+ * @param options options sent to customize the mutation
+ */
+const useUpdateMutationResultForPlugin = (
+  resultFromGraphql: MutationResult<any>, mutation: string, options?: object,
+) => {
+  const [projectionResult, setProjectionResult] = useState(projectMutationResult(resultFromGraphql));
+  const previousProjectionResult = usePreviousValue(projectionResult);
+
+  useEffect(() => {
+    const currentProjectionResult = projectMutationResult(resultFromGraphql);
+    if (!isEqual(currentProjectionResult, previousProjectionResult)) {
+      setProjectionResult(currentProjectionResult);
+    }
+  }, [resultFromGraphql]);
```

Comment:
This code snippet is triggering multiple dispatches of mutation result. This is caused because the initial value of `previousProjectionResult` is `undefined`. 
```suggestion
  const projection = projectMutationResult(resultFromGraphql);
  const [projectionResult, setProjectionResult] = useState(projection);

  useEffect(() => {
    setProjectionResult((prev) => {
      if (!isEqual(prev, projection)) return projection;
      return prev;
    });
  }, [resultFromGraphql]);
```